### ISSUE TITLE
Different results with feImage

### ISSUE THREAD

**Issue by afdw** [RazrFalcon/resvg#380]
Consider the following file:
```svg
<?xml version="1.0" encoding="UTF-8"?>
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="100" height="100" viewBox="0 0 100 100">
    <defs>
        <filter id="filter-0" x="0%" y="0%" width="100%" height="100%">
            <feImage xlink:href="#compositing-group-0"/>
        </filter>
        <g id="compositing-group-0">
            <circle cx="10" cy="10" r="30"/>
        </g>
    </defs>
    <g filter="url(#filter-0)">
        <rect x="-25" y="-10" width="100" height="30"/>
    </g>
</svg>
```

The results are:
| resvg | rsvg | Chrome, Inkscape |
| - | - | - |
| ![re](https://user-images.githubusercontent.com/7559560/116814649-a77c0680-ab62-11eb-9c44-6dbcb9537719.png) | ![r](https://user-images.githubusercontent.com/7559560/116814654-acd95100-ab62-11eb-817a-f79527dd4833.png) | ![ink](https://user-images.githubusercontent.com/7559560/116814795-2cffb680-ab63-11eb-916e-cdd5fd1c7d84.png) |

Who is right here? I do not seem to understand the SVG specification on how all these coordinates work.

Also, removing the `x="0%" y="0%" width="100%" height="100%"` part changes the result in Chrome and Inkscape, but not in resvg and rsvg.

**Comment by RazrFalcon** [comment]
There is no such thing as correctly rendered SVG. SVG doesn't have a reference implementation after all. So your results are not that surprising.

Usage of relative coordinates, like %, is bad idea in general. They are badly supported. Stick with absolute one.

Also, I'm not sure what result are you trying to achieve, so no help here either. Can you explain what result do you expect?

**Comment by afdw** [comment]
> There is no such thing as correctly rendered SVG. SVG doesn't have a reference implementation after all. So your results are not that surprising.

But there are specifications. If something important is not clear, I think they should be corrected.

I think the problem here arises from the fact that `feImage` has attributes `x`, `y`, `width`, `height` that are both [filter-primitive-attributes](https://drafts.fxtf.org/filter-effects/#filter-primitive-attributes) and `image`/`use` attributes somehow, and different implementations interpret them differently.
 
> Usage of relative coordinates, like %, is bad idea in general. They are badly supported. Stick with absolute one.

There is not much of relative coordinates here, and using absolute everywhere does not change the situation I think.

> Also, I'm not sure what result are you trying to achieve, so no help here either. Can you explain what result do you expect?

I want to compose "layers" (of arbitrary sizes and not necessarily with only positive coordinates) with different filters. See #381 for a real example of generated code (I think I have managed to make it work in rsvg and Chrome and sometimes in Inkscape).

**Comment by RazrFalcon** [comment]
[Filter primitive region](https://drafts.fxtf.org/filter-effects/#FilterPrimitiveSubRegion) is a very complicated and badly documented feature. You should not rely on it, because it's mostly broken in all implementations.
And since there are no reference implementation there is no correct answer.

**Comment by LaurenzV** [comment]
The output from `resvg` seems to match Chrome and Inkscape now, so I think this can be closed?

<img width="1403" alt="image" src="https://github.com/RazrFalcon/resvg/assets/47084093/8100cd2c-1f15-417c-8e5a-671c363aa54f">

![test](https://github.com/RazrFalcon/resvg/assets/47084093/39b8f6a5-8f4f-4a69-b364-70817e6820b2)

Probably fixed by the same commit as #381.

**Comment by RazrFalcon** [comment]
We should still add tests for both of those cases.
